### PR TITLE
Output cmdline/file diagnostics when args provided

### PR DESCRIPTION
**Pull request by uudisaru**
IB-4480: Create commandline application that can be started without GUI.
Add application class to check provided parameters and output the diagnostics to file or shell.
Signed-off-by: Toomas Uudisaru <toomas.uudisaru@aktors.ee>

### CODE REVIEW

File: CliApplication.cpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+{
+	this->outFile = outFile;
+}
+
+bool CliApplication::isDiagnosticRun()
+{
+	for( size_t i = 1; i < argc; ++i )
+	{
+		auto parameter = QString( argv[i] );
+		if( parameter.startsWith("/diag") )
```
Comment (martinpaljak) [inline CliApplication.cpp]: Do not use Windows conventions for command line parameters. Use dash (-).

Comment (uudisaru) [inline CliApplication.cpp]: Yes, that's actually fixed with the next commit https://github.com/open-eid/qt-common/pull/7/commits/acfcd20ff48bb7f95f1759bff42e682cc4d631f1

Comment (martinpaljak) [inline CliApplication.cpp]: Maybe just squash the commits then?

File: DiagnosticsTask.cpp [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+
+DiagnosticsTask::DiagnosticsTask( QObject *parent, const QString &appInfo, const QString &outFile )
+	: QObject(parent), appInfo(appInfo), outFile(outFile)
+{
+}
+
+void DiagnosticsTask::run()
+{
+	Diagnostics *worker = new Diagnostics( appInfo );
```
Comment (Telgat) [inline DiagnosticsTask.cpp]: Is this going to leak? (Maybe a scoped ptr instead)

File: CliApplication.cpp [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+	qtApp.setApplicationVersion( QString( "%1.%2.%3.%4" )
+		.arg( MAJOR_VER ).arg( MINOR_VER ).arg( RELEASE_VER ).arg( BUILD_VER ) );
+	qtApp.setOrganizationDomain( "ria.ee" );
+	qtApp.setOrganizationName( ORG );
+
+	QString appInfo;
+	QTextStream s( &appInfo );
+	diagnostics(s);
+
+	DiagnosticsTask *task = new DiagnosticsTask( &qtApp, appInfo, outFile );
```
Comment (Telgat) [inline CliApplication.cpp]: When will the task object created here be deleted?

**Pull request merged**